pkg/controller/deployment/deploment_helper_test.go

```diff
@@ -0,0 +1,398 @@
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("500m"),
+					corev1.ResourceMemory: k8sresource.MustParse("128Mi"),
+				},
+			},
+			expected: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("500m"),
+					corev1.ResourceMemory: k8sresource.MustParse("128Mi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("10m"),
+					corev1.ResourceMemory: k8sresource.MustParse("32Mi"),
+				},
+			},
+		},
+		{
+			name: "override resources doesn't replace source resource limits, replaces source requests",
+			sourceResources: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("400m"),
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("10m"),
+					corev1.ResourceMemory: k8sresource.MustParse("32Mi"),
+				},
+			},
+			overrideResources: v1alpha1.CertManagerResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("40m"),
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+			},
+			expected: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("400m"),
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("40m"),
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+			},
+		},
+		{
+			name: "override resources limits and requests merges with source resource limits and requests respectively",
+			sourceResources: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU: k8sresource.MustParse("10m"),
+				},
+			},
+			overrideResources: v1alpha1.CertManagerResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU: k8sresource.MustParse("400m"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+			},
+			expected: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("400m"),
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    k8sresource.MustParse("10m"),
+					corev1.ResourceMemory: k8sresource.MustParse("64Mi"),
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			actualResources := mergeContainerResources(tc.sourceResources, tc.overrideResources)
+			require.Equal(t, tc.expected, actualResources)
+		})
+	}
+}
+
+func TestGetOverrideResourcesFor(t *testing.T) {
```

Comment:
@thejasn @swghosh PLMK if this unit test is also redundant. I'll remove this one as well.

Reply:
I think we shouldn’t remove this one, but rather keep it. Cause, it effectively tests the switch case in the actual func that switches between different deployments (in the past we’ve seen a bug where we accidentally made mistakes in such For functions, so better to retain this).